### PR TITLE
Update Best Practices for Identifying Users page

### PR DESCRIPTION
**Pull request by sarahrudy**
### Proposed changes
- Update intro paragraph 
- Update AnonymousID generation blurb
- Add section on soft user registration 
- Add section on full user registration 
- Improve identifying users paragraph
- Add section on Linking server and client generated Ids
- Add situation to When to make identify call 
- Keep formatting of Segment methods consistent throughout page

### Merge timing
- ASAP once approved

### CODE REVIEW

**Comment by forstisabella** [comment]
Hi @sarahrudy - for some context on the delay in approval: we don't typically include images in the docs because they become unwieldy to maintain, but we are working on a "dummy" form so that customers to see the output of the identify and track calls when a user signs up for a newsletter, visits a website, etc. For visibility, the work is on this branch: https://github.com/segmentio/segment-docs/tree/isabella-dummy-form-test!

**Comment by sarahrudy** [comment]
@forstisabella ah, ok! thank you for letting me know! that's a great idea.

**Comment by sarahrudy** [comment]
@forstisabella - is there anything i'm able to help with here?

**Comment by forstisabella** [comment]
Hi @sarahrudy - we're still working on the dummy form, but I'll make sure that this goes out next week! Is it okay if I hide (comment out) the images temporarily?

**Comment by sarahrudy** [comment]
Hi @forstisabella, we could probably replace the images with code snippets, if that's easier!

**Comment by forstisabella** [comment]
I'll make that change! Thanks @sarahrudy

File: src/connections/spec/best-practices-identify.md [hunk 1]
```diff
@@ -3,64 +3,246 @@ title: Best Practices for Identifying Users
...
 
-## Identifying users
+Whenever possible, follow the `identify` call with a `track` event that records what caused the user to be identified.
 
-Segment's Identify method lets you link a user to their actions and record traits about them. It includes a unique User ID, and records any traits you know about them, such as their email address, and name.
+## AnonymousId generation
 
-Segment recommends that you use a unique user identifier that won't change for your `userId`, for example a database ID from your organization's internal systems. (See below)
+If you're using Segment's browser or mobile libraries, the Segment SDK generates and sets a UUID as `anonymousID` at the user's first presence. That `anonymousId` is saved in the user's cookie, as well as localStorage, and will stick with that user until the cache is cleared or a `reset` call is triggered.
```
Comment (forstisabella) [inline src/connections/spec/best-practices-identify.md:21]:
```suggestion
If you're using Segment's browser or mobile libraries, the Segment SDK generates and sets a UUID as `anonymousID` at the user's first visit to your site. That `anonymousId` is saved in the user's cookie, as well as localStorage, and will stick with that user until the cache is cleared or a `reset` call is triggered. 
```

**Review by forstisabella — APPROVED** [review]

File: src/connections/spec/best-practices-identify.md [hunk 2]
```diff
@@ -3,64 +3,244 @@ title: Best Practices for Identifying Users
...
+  "UserId": null
+}
+```
+
+<!--[No UserId in Payload](https://user-images.githubusercontent.com/78389005/214353985-599e7456-9295-4cb3-b97c-b30a9b905fcf.png)-->
+
+
+## Full User Registration 
+
+An anonymous visitor registers for an account and becomes a known user. The account creation process allows you to assign a `userId` from your production database, as well as capture additional traits. For this example, the `userId` that is assigned is "123abc". This is when you'll want to fire an `identify` call with this user's newly assigned `userId` and additional traits.
```
Comment (forstisabella) [inline src/connections/spec/best-practices-identify.md:165]:
```suggestion
An anonymous visitor registers for an account and becomes a known user. The account creation process allows you to assign a `userId` from your production database and capture additional traits. For this example, the `userId` that is assigned is "123abc". This is when you'll want to fire an `identify` call with this user's newly assigned `userId` and additional traits.
```

**Review by forstisabella — APPROVED** [review]

**Pull request merged**